Mathlib/Analysis/SpecialFunctions/Log/Summable.lean

```diff
@@ -8,47 +8,105 @@ import Mathlib.Analysis.SpecialFunctions.Complex.LogBounds
 import Mathlib.Analysis.NormedSpace.FunctionSeries
 
 /-!
-# Summable logs
+# Summability of logarithms
 
-We give conditions under which the logarithms of a summble sequence is summable. We also give some
-results about when the sums converge uniformly.
-
-TODO: Generalise the indexing set from ℕ to some arbitrary type, but this needs
-Summable.tendsto_atTop_zero to first be done. Also remove hff from
-`Complex.multipliable_one_add_of_summable` once we have vanishing/non-vanishing results for infinite
-products.
+We give conditions under which the logarithms of a summble sequence is summable. We also use this
+to relate summability of `f` to multipliability of `1 + f`.
 
 -/
 
-open Filter Function Complex Real
+variable {ι : Type*}
+
+open Filter Topology NNReal
 
-open scoped Interval Topology BigOperators Nat Complex
+namespace Complex
+variable {f : ι → ℂ} {a : ℂ}
 
-variable {ι : Type*}
+lemma hasProd_of_hasSum_log (hfn : ∀ i, f i ≠ 0) (hf : HasSum (fun i ↦ log (f i)) a) :
+    HasProd f (exp a) :=
+  hf.cexp.congr (by simp [exp_log, hfn])
+
+lemma multipliable_of_summable_log (hfn : ∀ n, f n ≠ 0) (hf : Summable fun i ↦ log (f i)) :
```

Comment:
Do you really need `hfn`? If `f n = 0` for some `n`, isn't the sequence trivially multipliable? If that's the case, you can remove the next lemma.

Reply:
Probably not. This code was not new in the present PR – it was just relocated from place to place – but I guess that's not a reason to ignore an opportunity to make it better.